packages/in_app_purchase/in_app_purchase_android/android/src/main/java/io/flutter/plugins/inapppurchase/MethodCallHandlerImpl.java

```diff
@@ -144,155 +97,107 @@ void setActivity(@Nullable Activity activity) {
-      case MethodNames.QUERY_PRODUCT_DETAILS:
-        List<Product> productList = toProductList(call.argument("productList"));
-        queryProductDetailsAsync(productList, result);
-        break;
-      case MethodNames.LAUNCH_BILLING_FLOW:
-        launchBillingFlow(
-            (String) call.argument("product"),
-            (String) call.argument("offerToken"),
-            (String) call.argument("accountId"),
-            (String) call.argument("obfuscatedProfileId"),
-            (String) call.argument("oldProduct"),
-            (String) call.argument("purchaseToken"),
-            call.hasArgument("prorationMode")
-                ? (int) call.argument("prorationMode")
-                : PRORATION_MODE_UNKNOWN_SUBSCRIPTION_UPGRADE_DOWNGRADE_POLICY,
-            result);
-        break;
-      case MethodNames.QUERY_PURCHASES_ASYNC:
-        queryPurchasesAsync((String) call.argument("productType"), result);
-        break;
-      case MethodNames.QUERY_PURCHASE_HISTORY_ASYNC:
-        queryPurchaseHistoryAsync((String) call.argument("productType"), result);
-        break;
-      case MethodNames.CONSUME_PURCHASE_ASYNC:
-        consumeAsync((String) call.argument("purchaseToken"), result);
-        break;
-      case MethodNames.ACKNOWLEDGE_PURCHASE:
-        acknowledgePurchase((String) call.argument("purchaseToken"), result);
-        break;
-      case MethodNames.IS_FEATURE_SUPPORTED:
-        isFeatureSupported((String) call.argument("feature"), result);
-        break;
-      case MethodNames.GET_CONNECTION_STATE:
-        getConnectionState(result);
-        break;
-      case MethodNames.GET_BILLING_CONFIG:
-        getBillingConfig(result);
-        break;
-      case MethodNames.IS_ALTERNATIVE_BILLING_ONLY_AVAILABLE:
-        isAlternativeBillingOnlyAvailable(result);
-        break;
-      case MethodNames.CREATE_ALTERNATIVE_BILLING_ONLY_REPORTING_DETAILS:
-        createAlternativeBillingOnlyReportingDetails(result);
-        break;
-      case MethodNames.SHOW_ALTERNATIVE_BILLING_ONLY_INFORMATION_DIALOG:
-        showAlternativeBillingOnlyInformationDialog(result);
-        break;
-      default:
-        result.notImplemented();
-    }
-  }
-
-  private void showAlternativeBillingOnlyInformationDialog(final MethodChannel.Result result) {
-    if (billingClientError(result)) {
+  public void showAlternativeBillingOnlyInformationDialog(
+      @NonNull Result<PlatformBillingResult> result) {
+    if (billingClient == null) {
+      result.error(getNullBillingClientError());
       return;
     }
     if (activity == null) {
-      result.error(ACTIVITY_UNAVAILABLE, "Not attempting to show dialog", null);
+      result.error(new FlutterError(ACTIVITY_UNAVAILABLE, "Not attempting to show dialog", null));
       return;
     }
-    billingClient.showAlternativeBillingOnlyInformationDialog(
-        activity,
-        billingResult -> {
-          result.success(fromBillingResult(billingResult));
-        });
+    try {
```

Comment:
I had to add all of these try/catch wrappers to all the `*Async` methods because it turns out the plugin is relying on them as part of its API 🙁. We have an integration test that checks that when we pass an empty product query, for instance, that we get back a platform exception (with a specific string, even though we don't control that string, so that could be fun someday).

Ideally we should be preventing invalid cases explicitly instead, but that would change behavior. These `FlutterError`s match the generic error made by the method channel code in the engine, so it's doing the same thing, it's just more explicit now (which means more boilerplate than pre-Pigeon, but also that we can see the anti-pattern—and the need to fix it—more clearly).